runtime/lua/vim/lsp.lua

```diff
@@ -1301,7 +1302,8 @@ nvim_command("autocmd VimLeavePre * lua vim.lsp._vim_exit_handler()")
 ---
 ---@param bufnr (number) Buffer handle, or 0 for current.
 ---@param method (string) LSP method name
----@param params (optional, table) Parameters to send to the server
+---@param params (optional, table | funtion number -> table) Parameters to send to the server, or a function
```

Comment:
I don't like that we've made params polymorphic

Reply:
Me neither. What do you think about my idea above about making a more general function that _only_ takes a function?

Reply:
I think if it was something like a utility function that set up the params (and then dispatched) each request with the appropriate positional params (where params was still of the old type), that would work. Generally I want to avoid introducing more middleware like buf_request/new types of requests like pos_request.

Reply:
So, you'd rather precompute the params rather than getting them between client requests, do I understand correctly? And maybe pass them in to some function as a mapping from client to params?